app/views/facilities/_quicklinks.html.haml

```diff
@@ -0,0 +1,21 @@
+.quick_links 
+  .add_to_cart
+    - if not session_user
+      = link_to "#{t("admin.shared.add_to",model:t("pages.cart"))} (#{t("pages.login")})", new_user_session_path
+    - elsif session_user.can_override_restrictions?(product) || product.can_be_used_by?(acting_user)
+      - if not session_user.accounts.empty?
+        = link_to t("admin.shared.add_to",model:t("pages.cart")), add_order_path(acting_user.cart(session_user), order: {order_details: [{product_id: product.id, quantity: 1}] }), method: :put
+      - else
+        = t("facility_accounts.account_table.notice")
+    - else
+      = t("facility_account_orders.cannot_purchase")
+  %li{class: [product.class.model_name.to_s.downcase, 'view_product']}
+    = public_calendar_link(product, indent: indent_for_icon)
+    = link_to 'View' + (product.is_hidden? ? ' (hidden)' : ''), facility_product_path(current_facility || product.facility, product), :id => product.name
+    - if acting_user.present? && !product.can_be_used_by?(acting_user)
+      %i.fa.fa-lock
+      = " (#{product.class.human_attribute_name(:requires_approval_show)})"
+
+      - if product.offline?
+        = tooltip_icon "fa fa-exclamation-triangle icon-large", t("instruments.offline.note")
+    
```

Comment:
Can you enable your text editor to automatically add a blank newline to the end of files? In sublime the setting is `"ensure_newline_at_eof_on_save": true`,